src/Uri.php

```diff
@@ -576,6 +576,26 @@ private function filterScheme($scheme)
         return strtolower($scheme);
     }
 
+    /**
+     * @param string $component
+     *
+     * @return string
+     *
+     * @throws \InvalidArgumentException If the user info is invalid.
+     */
+    private function filterUserInfoComponent($component)
```

Comment:
please also call this method from `applyParts` and add a test of this when the user info is set from the uri constructor

Reply:
There's already tests for `applyParts`. Should I modify them to include special characters in `userInfo`, or create new tests?